client/app/stores/message_store.coffee

```diff
@@ -606,13 +606,18 @@ class MessageStore extends Store
         if param.type is 'message' and not _conversationMemoize.size
             messages = _currentMessages
 
-        getMessage = ->
-            index0 = messages.toArray().findIndex (message) ->
-                if param.conversationID isnt 'undefined'
-                    return param.conversationID is message.get 'conversationID'
-                return messageID is message.get 'id'
-            index = param.transform index0
-            messages.get index
+        getMessage = =>
+            _newMessage = (index) ->
```

Comment:
On the other hand, if it fixes a bug and we actually do the big refactoring(TM), whatever ;-)